### PR TITLE
Patch to improve satellite(foreman) dynamic inventory script performance

### PR DESCRIPTION
**Pull request by skamithi**
##### SUMMARY
Improves Satellite inventory script by running API calls asynchronously instead of synchronously. 

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
contrib/inventory/foreman.py

##### ANSIBLE VERSION

```
ansible 2.5.0
  config file = None
  configured module search path = [u'/home/linuxsimba/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/linuxsimba/tower-cli/local/lib/python2.7/site-packages/ansible
  executable location = /home/linuxsimba/tower-cli/bin/ansible
  python version = 2.7.12 (default, Nov 19 2016, 06:48:10) [GCC 5.4.0 20160609]
```

##### ADDITIONAL INFORMATION
Improves performance by 50%. Tested on a Satellite server configured to manage 600 Hosts.
Just works with Ansible Tower . Outside of Tower requires the ``futures`` python library.  An assertion is made to ensure the ``futures`` library is accesible.

time taken before the patch
```
real    3m8.531s
user    0m2.063s
sys     0m0.507s
```

time taken after the patch
```
real    1m41.734s
user    0m3.143s
sys     0m0.577s
```

### CODE REVIEW

File: contrib/inventory/foreman.py [hunk 1]
```diff
@@ -46,6 +46,13 @@
 
 from requests.auth import HTTPBasicAuth
 
+try:
+    import concurrent.futures
```
Comment (maxamillion) [inline contrib/inventory/foreman.py:50]: My main concern with this is that `futures` is not available in base RHEL, Satellite, or Ansible Engine subscription channels for Red Hat Customers. The only channels I see it currently available in is OpenStack for EL6 and both OpenStack and Gluster for EL7, both which are add-on SKUs and not available to customers who haven't bought those other products..

From a community upstream standpoint, this is perfectly fine. However, if the intent is for this to be used with the product version of Satellite (which I assume it is given the use of the product name in the PR) then that's problematic.

Comment (bcoca) [reply contrib/inventory/foreman.py:50]: also, this script is on it's way out, use the foreman inventory plugin instead

**Comment by ekohl** [comment]
wontfix since we'd like to focus on the inventory script.

**Comment by bcoca** [comment]
closing as per above

**Comment by ekohl** [comment]
I thought the bot would pick that up. Did I do it wrong or did it need time to process?

**Comment by bcoca** [comment]
nothing wrong, I just happen to fly by before bot got to it

**Pull request closed without merging**